janusgraph-test/src/main/java/org/janusgraph/diskstorage/indexing/IndexProviderTest.java

```diff
@@ -1057,6 +1057,63 @@ public void testScroll() throws BackendException {
         assertEquals(6, tx.queryStream(new IndexQuery(store, PredicateCondition.of(NAME, Cmp.EQUAL, "Hello world"), 6)).count());
     }
 
+    @Test
+    public void testPersistantIndexData() throws BackendException {
+        final String store = "vertex";
+
+        final Multimap<String, Object> initialDoc = HashMultimap.create();
+        initialDoc.put(DATE, Instant.ofEpochSecond(1));
+        double latitude = random.nextDouble() * 180 - 90;
+        double longitude = random.nextDouble() * 360 - 180;
+        initialDoc.put(LOCATION, Geoshape.point(latitude, longitude));
+
+        initialDoc.put(STRING, "network");
+        initialDoc.put(ANALYZED, "network");
+        initialDoc.put(KEYWORD, "network");
+        //initialDoc.put(FULL_TEXT, "network");
+        initialDoc.put(NAME, "network");
+        initialDoc.put(TIME, 1L);
+        initialDoc.put(WEIGHT, 1.5d);
+        Date date = new Date(1L);
+
+        if (indexFeatures.supportsCardinality(Cardinality.LIST)) {
+            initialDoc.put(PHONE_LIST, "one");
+            initialDoc.put(PHONE_LIST, "two");
+        }
+        if (indexFeatures.supportsCardinality(Cardinality.SET)) {
+            initialDoc.put(PHONE_SET, "three");
+            initialDoc.put(PHONE_SET, "four");
+        }
+
+        initialize(store);
+        add(store, "doc1", initialDoc, true);
+
+        clopen();
+
+        // update the document
+        tx.add(store, "doc1", TEXT, "update", false);
+        tx.commit();
+
+        // check every indexed type is still in the index
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(DATE, Cmp.EQUAL, Instant.ofEpochSecond(1)))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(LOCATION, Geo.WITHIN, Geoshape.box(latitude-1L, longitude-1, latitude+1, longitude+1)))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(WEIGHT, Cmp.EQUAL, 1.5d))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(TIME, Cmp.EQUAL, 1L))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(STRING, Cmp.EQUAL, "network"))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(KEYWORD, Text.CONTAINS, "network"))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(ANALYZED, Text.CONTAINS, "network"))).count());
+        assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(NAME, Cmp.EQUAL, "network"))).count());
+
```

Comment:
I see that you partially copied the test from #929 but you removed the next lines from the test:
```
assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(BOOLEAN, Cmp.EQUAL, true))).count());
assertEquals(1, tx.queryStream(new IndexQuery(store, PredicateCondition.of(TRUEDATE, Cmp.EQUAL, date))).count());
```

Is there a reason why you didn't include these assertions?

Reply:
TRUEDATE and BOOLEAN indexed is same as DATE, NUMERIC for test simplicity I removed thats fields

Reply:
Than `Date date = new Date(1L);` isn't needed as suggested by Codacy

Reply:
Done